src/charm.py

```diff
@@ -55,29 +65,29 @@ def __init__(self, *args):
             cluster_name=str(),
         )
 
-        self._db = MySQLClient(self, "db")
+        self._slurmdbd_manager = SlurmdbdManager()
+        self._db = DatabaseRequires(self, relation_name="database", database_name="slurm_acct_db")
         self._slurm_manager = SlurmManager(self, "slurmdbd")
         self._slurmdbd = Slurmdbd(self, "slurmdbd")
         self._slurmdbd_peer = SlurmdbdPeer(self, "slurmdbd-peer")
         self._fluentbit = FluentbitClient(self, "fluentbit")
 
-        event_handler_bindings = {
+        for event, handler in {
             self.on.install: self._on_install,
             self.on.upgrade_charm: self._on_upgrade,
             self.on.update_status: self._on_update_status,
             self.on.config_changed: self._write_config_and_restart_slurmdbd,
             self.on.jwt_available: self._on_jwt_available,
             self.on.munge_available: self._on_munge_available,
             self.on.write_config: self._write_config_and_restart_slurmdbd,
-            self._db.on.database_available: self._write_config_and_restart_slurmdbd,
-            self._db.on.database_unavailable: self._on_db_unavailable,
+            self._db.on.database_created: self._on_database_created,
+            self._db.on.endpoints_changed: self._on_database_endpoints_changed,
             self._slurmdbd_peer.on.slurmdbd_peer_available: self._write_config_and_restart_slurmdbd,
             self._slurmdbd.on.slurmctld_available: self._on_slurmctld_available,
             self._slurmdbd.on.slurmctld_unavailable: self._on_slurmctld_unavailable,
             # fluentbit
             self.on["fluentbit"].relation_created: self._on_fluentbit_relation_created,
-        }
-        for event, handler in event_handler_bindings.items():
+        }.items():
```

Comment:
Nice, much easier to read now

Reply:
One-time use variables and if-else chains fear me :smiling_imp: